src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -331,26 +326,16 @@ public boolean waitForReferenceProcessing()
             public void runFinalization() {
                 Finalizer.runFinalization();
             }
-        });
-
-        referenceHandlerResource = new JDKResource() {
-            @Override
-            public Priority getPriority() {
-                return Priority.REFERENCE_HANDLER;
-            }
 
             @Override
-            public void beforeCheckpoint(Context<? extends Resource> context) throws Exception {
-                System.gc();
-                // TODO ensure GC done processing all References
-                while (waitForReferenceProcessing());
-            }
-
-            @Override
-            public void afterRestore(Context<? extends Resource> context) throws Exception {
+            public boolean waitForQueueProcessed(ReferenceQueue<?> queue,
```

Comment:
This is an interface method of JavaLangRefAccess that exposes some package-private java.lang.ref methods. So it cannot be static. `this` in this context is the instance of that JavaLangRefAccess. And the interface methods just calls the instance method of ReferenceQueue as you suggest.

That instance method is intentionally made not-public, so our EA code will be compatible with JDK17. After some thought, this looks like better approach than introducing new public methods until those methods will be agreed to be good ones. I don't feel this for the method, after all, I found a better name since the last patch in this area :)

Reply:
Reading this again, I now see the new method is a member of the `new JavaLangRefAccess() {}` class which makes perfect sense for why it's here like this.